modules/protocol/resources/META-INF/smithy/smithy4s.meta.smithy

```diff
@@ -13,3 +13,7 @@ namespace smithy4s.meta
 
 @trait(selector: ":is(service, operation)")
 structure packedInputs {}
+
+@trait(selector: "structure")
+@idRef(failWhenMissing: true, selector: "union")
+string adtMember
```

Comment:
we should really add documentation directly inlined in the smithy spec as to what those do